cmd/cli/certificate_rotate.go

```diff
@@ -0,0 +1,352 @@
+			if err != nil {
+				return fmt.Errorf("could not access Kubernetes cluster, check kubeconfig: %w", err)
+			}
+			rotate.configClient = configClient
+
+			clientset, err := kubernetes.NewForConfig(config)
+			if err != nil {
+				return fmt.Errorf("could not access Kubernetes cluster, check kubeconfig: %w", err)
+			}
+			rotate.clientSet = clientset
+			rotate.prompter = fmt.Scanln
+
+			return rotate.run()
+		},
+	}
+
+	f := cmd.Flags()
+
+	f.StringVar(&rotate.meshName, "mesh-name", defaultMeshName, "Name of the service mesh")
+	f.StringVarP(&rotate.newTrustDomain, "trust-domain", "t", "", "If specified the new cert will use this trust domain. Works with certificate provider tresor")
+	f.BoolVarP(&rotate.prompt, "accept", "y", false, "when specified it will not prompt for user input")
+	f.BoolVarP(&rotate.deleteOld, "delete", "d", false, "when specified it will delete the old MeshRootCertificate, otherwise the MeshRootCertificate will remain as inactive")
+	f.DurationVarP(&rotate.waitForRotation, "wait", "w", 60*time.Second, "Time to wait for certificate to propagate to all components at each step")
+	f.StringVarP(&rotate.mrcFilePath, "file", "f", "", "File to use for the new MRC. When not supplied the settings from the existing MRC are copied")
+	f.StringVarP(&rotate.mrcName, "meshrootcertificate", "c", "", "Existing MeshRootCertificate to rotate to active role using the name in currently configured namespace")
+	return cmd
+}
+
+func (r *rotateCmd) promptForContinue(msg string) bool {
+	if !r.prompt {
+		var rotate string
+		fmt.Fprintf(r.out, msg+" y/n:")
+		_, err := r.prompter(rotate)
+		if err != nil {
+			fmt.Printf("warning: error reading value from user: %s\n", err)
+			return true
+		}
+
+		if rotate != "y" {
+			return false
+		}
+	}
+
+	return true
+}
+
+func (r *rotateCmd) run() error {
+	// TODO(#5204): check if MRC feature is enabled
+	// TODO(#4891): check cert rotations or metrics instead of using time
+	if !r.promptForContinue("Are you sure you want to initiate rotation?") {
+		return nil
+	}
+
+	oldMrc, err := r.findCurrentInUse()
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(r.out, "Found MeshRootCertificate [%s] for mesh [%s] in namespace [%s]\n\n", oldMrc.Name, r.meshName, settings.Namespace())
+
+	newMrc, err := r.createOrFindNewMRC(oldMrc)
+	if err != nil {
+		return fmt.Errorf("unable to create or find new MeshRootCertificate: %s", err)
+	}
+	fmt.Fprintf(r.out, "Created new MeshRootCertificate [%s] in Inactive role\n", newMrc.Name)
+
+	fmt.Fprintf(r.out, "Moving MeshRootCertificate [%s] to Passive role\n", newMrc.Name)
+	if !r.promptForContinue("Are you sure you want to initiate rotation?") {
+		return nil
+	}
+	err = r.updateCertificate(newMrc.Name, v1alpha2.PassiveIntent)
+	if err != nil {
+		return fmt.Errorf("unable to update MeshRootCertificate [%s] with role [%s]: %s", newMrc.Name, v1alpha2.PassiveIntent, err)
+	}
+	fmt.Fprintf(r.out, "waiting for propagation...\n\n")
```

Comment:
On all of the sleeps I would add "waiting %s for propoagation", r.waitForRotation so they know how long we're waiting